docs/product-user-guide/src/main/asciidoc/guided-rules-THEN-proc.adoc

```diff
@@ -1,31 +1,45 @@
 +
-The *Add a new action...* dialog window with the available action templates opens.
+The *Add a new action* window with the available action elements opens.
 +
-.Nurse Roster THEN Dialog Window
-image::5695.png[]
-. Choose an action (for example *Modify a ...*) and click *Ok*.
-. Click the newly added `THEN` action.
+.Add a new action to the rule
+image::5695.png[Add a new action to the rule]
 +
-An *Add a field* dialog appears. In this example, the field is for the "*Modify value of LoanApplication [a]*" image:6191.png[] action.
+The list includes insertion and modification options based on the data objects in the *Data Objects* tab of the guided rule designer, and on any DSL objects defined for the package (if you selected *Use Domain Specific Language (DSL)* when you created this guided rule):
+
+* *Change field values of:* Use this to set the value of fields on a fact (such as `LoanApplication`).
+* *Delete:* Use this to delete a fact.
+* *Modify:* Use this to specify fields to be modified for a fact.
+* *Insert fact:* Use this to insert a fact and define resulting fields and values for the fact.
+* *Logically Insert fact:* Use this to insert a fact logically into the inference engine and define resulting fields and values for the fact. The Red Hat Decision Manager inference engine is responsible for logical decisions on insertions and retractions of facts. After regular or stated insertions, facts have to be retracted explicitly. After logical insertions, facts are automatically retracted when the conditions that originally asserted the facts are no longer true.
+* *Add free form DRL:* Use this to insert a free-form DRL field where you can define condition elements freely, without the guided rule designer.
+* *Call method on:* Use this to invoke a method from another fact.
 +
-.Add a Field Dialog
-image::5696.png[An Add a field dialog for the Guided Template Editor in BRMS User Guide 6.0.2]
-. Within this dialog, choose a field from the *Add field* drop-down menu.
+*<@Michael (2 of 6): Same template key question, for free form DRL in Actions.>*
 +
-Once you select a field, the dialog window closes automatically.
-
-. Select the *Edit* icon (image:6191.png[]) within the item field to define the field value with a literal value, template key, or a formula.
-. Under the *THEN* section, click *(show options...)* and click the plus icon (image:5686.png[]) on the right side to set additional rule attributes, such as `ruleflow-group`, `salience`, and `date-expires`.
+////
+.Template keys in free form DRL
+[NOTE]
+====
+With free form DRL, you have the flexibility to add custom data types, such as template keys. Template keys are placeholder variables in your field constraints that are interpolated with a specified value when the DRL rules are generated. You can add a template key value to facts in the format `@key`. For example, `Person( age > @{age} )` creates an `$age` column in the decision table.
```

Comment:
Template Keys in "Guided Rules" (as opposed to "Guided Rule Templates") has no effect and will be treated as a literal value.